File: pset6/cash/cash.py
# Ruchella Kock
# 12460796
# this program will give the least amount of coins that can be given
from cs50 import get_float

while True:
    dollars = get_float("Change owed: ")
    coins = round(dollars * 100)
    # make sure you have a positive amount of change owed if not ask again
    if dollars < 0:
        continue
    else:
        break
change = 0
# determine how many quarters fit in coins
while coins >= 25:
    coins = coins - 25
    change = change + 1

# determine how many dimes fit in coins
while coins >= 10:
    coins = coins - 10
    change = change + 1

# determine how many quarters fit in coins
while coins >= 5:
    coins = coins - 5
    change = change + 1

# determine how many penny's fit in coins
change = change + coins

print(f"{change}")

File: pset6/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    int coins;
    //Prompt user for change needed and make sure its a value bigger than 0
    //Enters a value smaller than 0 ask again for a new value
    do
    {
        dollars = get_float("Change owed: ");
        coins = round(dollars * 100);
    }
    while (coins < 0);

    int modulo_25 = (coins % 25);
    int modulo_25_10 = (coins % 25 % 10);

    //First if statement will only print the amount of coins needed, if the amount of change can be divided by 25
    if (modulo_25 == 0)
    {
        printf("%i \n", coins / 25);
    }
    //Second if statement will only print the amount of coins needed, if the amount of change can be divided by 25 and 10
    else if (coins % 25 % 10 == 0)
    {
        printf("%i \n", ((coins - modulo_25) / 25) + ((modulo_25) / 10));
    }
    //Third if statement will only print the amount of coins needed, if the amount of change can be divided by 25,10 and 5
    else if (coins % 25 % 10 % 5 == 0)
    {
        printf("%i \n", ((coins - modulo_25) / 25) + ((modulo_25 - modulo_25_10) / 10) + ((modulo_25_10) / 5));
    }
    //Fourth if statement will only print the amount of coins needed, if the amount of change can be divided by 25,10, and 1
    else if (coins % 25 % 10 % 5 % 1 == 0)
    {
        printf("%i \n", ((coins - modulo_25) / 25) + ((modulo_25 - modulo_25_10) / 10) + ((modulo_25_10 - coins % 25 % 10 % 5) / 5) + ((
                    coins % 25 % 10 % 5) / 1));
    }
}